src/main/java/com/openshift/jenkins/plugins/pipeline/MessageConstants.java

```diff
@@ -0,0 +1,60 @@
+public static final String START_DEPLOY_CANCEL = "\n\nStarting the \"%s\" action for deployment config \"%s\" from the project \"%s\".";
+public static final String DEPLOY_NO_CFG = "\n\nExiting \"%s\" unsuccessfully; the deployment config \"%s\" could not be retrieved.";
+public static final String DEPLOY_CANCEL_NOOP = "\n\nExiting \"%s\" successfully; the deployment \"%s\" is not in-progress; its status is:  [%s].";
+public static final String DEPLOY_CANCEL_SUCCESS = "\n\nExiting \"%s\" successfully; the deployment \"%s\" has been cancelled.";
+public static final String DEPLOY_NO_REP = "\n\nExiting \"%s\" unsuccessfully; the latest deployment \"%s\" could not be retrieved.";
+public static final String START_DEPLOY = "\n\nStarting the \"%s\" step with deployment config \"%s\" from the project \"%s\".";
+public static final String DEPLOY_FAILED = "\n\nExiting \"%s\" unsuccessfully; deployment \"%s\" has completed with status:  [Failed].";
+public static final String DEPLOY_TIMED_OUT = "\n\nExiting \"%s\" unsuccessfully; gave up on deployment \"%s\" with status:  [%s].";
+public static final String DEPLOY_SUCCESS = "\n\nExiting \"%s\" successfully; deployment \"%s\" has completed with status:  [Complete].";
+public static final String DEPLOY_VERIFY_STATE = "  Verifying deployment state%s...";
+public static final String DEPLOY_VERIFY_REPLICAS = " and verifying the current deployment is at \"%s\" replica(s).";
+public static final String SCM_CALC = "\n\nThe \"%s\" SCM will return the last revision state stored in Jenkins for the image stream \"%s\" and tag \"%s\" from the project \"%s\".";
+public static final String SCM_LAST_REV = "  Last revision:  [%s]";
+public static final String SCM_NO_REV = "  No revision state has been retrieved and stored yet.";
+public static final String SCM_COMP = "\n\nThe \"%s\" SCM is pulling the lastest revision state from OpenShift for the image stream \"%s\" and tag \"%s\" from the project \"%s\" and storing in Jenkins.";
+public static final String SCM_CHANGE = "\n\n A revision change was found this polling cycle.";
+public static final String SCM_NO_CHANGE = "\n\n No revision change found this polling cycle.";
+public static final String START_TAG = "\n\nStarting the \"%s\" step with the source [image stream:tag] \"%s:%s\" and destination [image stream:tag] \"%s:%s\" from the project \"%s\".";
+public static final String EXIT_OK = "\n\nExiting \"%s\" successfully.";
+public static final String SCALING = "  Scaling to \"%s\" replicas%s...";
+public static final String SCALING_VERIFYING = " and verifying the replica count is reached.";
+public static final String SCALING_NOOP = "\n\nExiting \"%s\" successfully; no deployments for \"%s\" were found, so a replica count of \"0\" already exists.";
+public static final String SCALING_FAILED = "\n\nExiting \"%s\" unsuccessfully; the call to \"%s\" failed.";
```

Comment:
in jenkins general, there is Logger tracing

for the plugin, yes, each step has a verbose switch that has trace points, etc.